### PR TITLE
fix: better error messages for frames

### PR DESCRIPTION
**Pull request by JFrankfurt**
This has each frame track its own errors, whereas before all errors were aggregated at the context level (dumb).

Errors quickly fade in, and fade out when clicked. 

|before|after|
|-|-|
|image|image|

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/src/components/Basenames/UsernameProfileSectionFrames/Frame.tsx [hunk 1]
```diff
@@ -28,6 +59,51 @@ export default function Frame({ url, className }: FrameProps) {
     });
   };
 
+  const useSharedCallback = <T extends SignatureData | TransactionData>(
```
Comment (kirkas) [inline apps/web/src/components/Basenames/UsernameProfileSectionFrames/Frame.tsx:62]: This function is v hard to parse/read, maybe add comment or split?

**Review by kirkas — APPROVED** [review]
